conans/client/cmd/download.py

```diff
@@ -18,7 +18,7 @@ def download(reference, package_ids, remote_name, recipe, registry, remote_manag
 
     # First of all download package recipe
     remote_manager.get_recipe(reference, remote)
-    registry.set_ref(reference, remote)
+    registry.set_ref(reference, remote.name)
```

Comment:
I would say that the registry receives and returns Remote objects, not str. It is the remote_registry.py file the one the defines the ``Remote`` object.

Reply:
In the registry class there are several functions that receives "remote_names" because is the value that the user is passing. It would be very weird to ask to the registry "what is the remote for this name?" and then "ok, now save the reference for the object I just asked"